proto/opamp.proto

```diff
@@ -802,6 +802,22 @@ message PackageStatus {
 
     // Error message if the status is erroneous.
     string error_message = 7;
+
+    // Optional details that may be of interest to a user.
+    // Should only be set if status is Downloading.
+    // Status: [Development]
+    PackageDownloadDetails download_details = 8;
+}
+
+
+// Additional details that an agent can use to describe an in-progress package download.
+// Status: [Development]
+message PackageDownloadDetails {
+    // The package download progress as a percentage.
+    double download_percent = 1;
+
+    // The current package download rate in bytes per second.
+    uint64 download_rate = 2;
```

Comment:
I would prefer to include the unit in the name of the field. I think we could remove rate as it will be clear that it is a rate.

```suggestion
    // The current package download rate in bytes per second.
    uint64 download_bytes_per_second = 2;
```